flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/util/AwsV2Util.java

```diff
@@ -248,4 +254,29 @@ static AwsCredentialsProvider getWebIdentityTokenFileCredentialsProvider(
 	public static Region getRegion(final Properties configProps) {
 		return Region.of(configProps.getProperty(AWSConfigConstants.AWS_REGION));
 	}
+
+	/**
+	 * Whether or not an exception is recoverable.
+	 */
+	public static boolean isRecoverableException(ExecutionException e) {
+		if (!(e.getCause() instanceof SdkException)) {
+			return false;
+		}
+		SdkException ase = (SdkException) e.getCause();
+		return ase instanceof LimitExceededException || ase instanceof ProvisionedThroughputExceededException || ase instanceof ResourceInUseException;
```

Comment:
@xiaolong-sn `ase instanceof ResourceInUseException` can be replaced with `isResourceInUse(ase)`

Reply:
I re-read the code and found out that the ResourceInUseExeption might not be a recoverable exception and removed it.